tests/test_build.py

```diff
@@ -67,9 +67,8 @@ def test_build_html(sphinx_build_factory, file_regression):
     navbar = index_html.select("div#navbar-center")[0]
     file_regression.check(navbar.prettify(), basename="navbar_ix", extension=".html")
 
-    # Sidebar structure
-    sidebar = index_html.select(".bd-sidebar")[0]
-    file_regression.check(sidebar.prettify(), basename="sidebar_ix", extension=".html")
+    # Sidebars should be hidden on index page because there is no sub-page
+    assert not index_html.select(".bd-sidebar")
```

Comment:
```suggestion
    assert not index_html.select(".bd-sidebar-primary")
```

As you changed it in `sidebar-primary.htm ?